patterns/ink/InkMetadataRune.ts

```diff
@@ -89,7 +89,30 @@ export class InkMetadataRune<out U> extends Rune<InkMetadata, U> {
     chain: ChainRune<C, U>,
     props: RunicArgs<X, InstantiateProps>,
   ) {
-    const { code } = props
+    const code = Rune
+      .resolve(props.code)
+      .unhandle(undefined)
+      .rehandle(undefined, () =>
+        this.into(ValueRune)
+          .access("source", "wasm")
+          .unhandle(undefined)
+          .map(hex.decode)
+          .rehandle(undefined))
+    const codeHash = this.into(ValueRune).access("source", "hash").map(hex.decode)
+    // TODO: verify deployed or throw
+    // const sourceHash = this.into(ValueRune).access("source", "hash").map(hex.decode)
+    // const codeHash = Rune.tuple([code, sourceHash])
+    //   .map(([code, sourceHash]) =>
+    //     code
+    //       ? sourceHash
+    //       : chain.pallet("Contracts").storage("CodeStorage").valueRaw(
+    //         sourceHash as $.Native<Chain.Storage<C, "Contracts", "CodeStorage">["key"]>,
+    //       )
+    //         .unhandle(undefined)
+    //         .map(() => sourceHash)
+    //         .rehandle(undefined, () => Rune.constant(new NotDeployedError()))
+    //         .unhandle(NotDeployedError)
+    //   )
```

Comment:
@tjjfvi I'd like to throw a `NotDeployedError` if the contract has not been deployed.
How would you branch this logic with `Rune`s?
Right now, the logic assumes that there is a contract deployed for the given source hash.